src/Control/HTTPRequest.php

```diff
@@ -138,13 +144,185 @@ public function __construct($httpMethod, $url, $getVars = array(), $postVars = a
     {
         $this->httpMethod = strtoupper(self::detect_method($httpMethod, $postVars));
         $this->setUrl($url);
-
         $this->getVars = (array) $getVars;
         $this->postVars = (array) $postVars;
         $this->body = $body;
     }
 
     /**
+     * Create HTTPRequest instance from the current environment variables.
```

Comment:
This is feels like the "lots of static methods representing something that should be a separate class" anti-pattern. Maybe you want an `HTTPRequestBuilder` or `HTTPRequestFactory` here?